src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -1037,9 +1060,11 @@ private String key() {
         }
         @Override public FlowExecution getOrNull() {
             try {
-                ListenableFuture<FlowExecution> promise = run().getExecutionPromise();
+                WorkflowRun run = run();
+                ListenableFuture<FlowExecution> promise = run.getExecutionPromise();
                 if (promise.isDone()) {
-                    return promise.get();
+                    // Weird, I know, but this ensures we trigger the onLoad for the execution via the lazy-load mechanism
```

Comment:
So IIUC this would benefit historical callers or `getOrNull` which might try the call again shortly, such as GUI elements.